### ISSUE TITLE
CU - Link Publish Workflow to Ticket

### ISSUE THREAD

**Issue by TimRoe** [department-of-veterans-affairs/va-mobile-library#105]
## Proposed Change

It is good to have a paper trail of what ticket created what published NPM version--particularly for non-alpha builds. This ticket is to update our Publish GHA to add a comment to relevant ticket (preferably the main, but PR if needed) denoting the version creation so there's an automatic paper trail that can be followed if needed.

## Why Should We Prioritize?

> [S]o there's an automatic paper trail that can be followed if needed.

For example a meaningful bug is found and in tracking down where in the code the bug is we can easily know what package version it was added in and thereby all the versions that were put out with it broken. Also could be used for general backtracking purposes.

## Coding Time Estimation

3; less than a day
	
## Testing Considerations

None besides draft testing it works.

**Comment by TimRoe** [comment]
Looked into this ticket and, after discussions with Narin, we decided now was not the right time for it. 

tl;dr justification: we have a workaround and automatically commenting on issues with current processes is much more difficult than initially believed.

The ticket will not be closed though as the task itself still has merit. An appropriate time to revisit this ticket may be when the design system reaches a level of maturity where we move to publishing major versions for breaking changes instead of our temporary/learning/growing process of minor bumps potentially including breaking changes.

Full justification:
- There is a workaround for finding the paper trail if needed from the code (as one would look at for a bug): the most likely jumping off point is the git-blame log that tracks prior issues that touched the relevant line(s) of code which will point to a PR and all PRs (with the "changelog" label, which all touching package-relevant source code should have) will be documented in the [Changelog file](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/CHANGELOG.md), thereby allowing finding the version with a simple `Ctrl + F` and putting in the PR issue number
- Since this ticket was created, our process shifted from publishing packages on the branch to doing so from main (so the Changelog would work correctly) which complicates things as a package publication is no longer inherently tied to a ticket
   - For example: we have previously intentionally stacked up multiple PRs for a complex component and only later published lumping in multiple tickets so we'd effectively have to search every commit since last publication (a boundary hard to ascertain) to get issue numbers to comment on
- Currently we do not have any good way to get the issue numbers from commits to main, so it'd require string parsing the commit messages and string parsing is always brittle
- Currently we do not delineate packages being published in any capacity at the issue/PR/commit level so there's no way to certifiably know that the comments would be accurate (it was noted this recently was also inaccurate in the Changelog, Narin noted now that we're in our own repo we could make labels to denote the packages to remedy by either splitting to have a Changelog for each package or using labels for which package a PR updates to refine the Changelog processing)
- A potential option was somehow integrating the process with the Changelog generation which already parses out a list of tickets since last publication, but looking at what we're doing it seems that behavior is contained within a library we're leveraging for generating it and not our logic

Some resource notes from digging around into how to implement:
- [Commenting on a PR with GHA (GitHub Actions)](https://stackoverflow.com/questions/58066966/commenting-a-pull-request-in-a-github-action)
   - [GitHub doc within link](https://github.com/actions/github-script#comment-on-an-issue)
- [Getting commit messages in GHA](https://stackoverflow.com/questions/63619329/how-to-get-the-commit-message-in-github-actions)
   - [GitHub doc within link](https://docs.github.com/en/webhooks/webhook-events-and-payloads#push)
   - [More GitHub doc within link](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)
- [Commits to main of library repo](https://github.com/department-of-veterans-affairs/va-mobile-library/commits/main/)
   - Example commit message string we'd have to parse for ticket numbers (PR or manually added main ticket in branch name): `Merge pull request #360 from department-of-veterans-affairs/feature/263-narin-theme-setup-tokens`
- [Publish script containing generating the changelog](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/.github/workflows/publish.yml)
   - [Changelog script in our logic](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/.github/scripts/generate-changelog.sh)
   - [Resulting Changelog](https://github.com/department-of-veterans-affairs/va-mobile-library/blob/main/CHANGELOG.md) that contains PRs merged that contributed to the version (including PR issue numbers for manually connecting the dots either from the version or the code via git-blame pointing you to the PR)
